notary-plugins/notary-plugin-non-validating/notary-plugin-non-validating-server/src/test/kotlin/com/r3/corda/notary/plugin/nonvalidating/server/NonValidatingNotaryServerFlowImplTest.kt

```diff
@@ -0,0 +1,299 @@
+package com.r3.corda.notary.plugin.nonvalidating.server
+import com.r3.corda.notary.plugin.common.InternalNotaryException
+import com.r3.corda.notary.plugin.common.NotarisationRequestSignature
+import com.r3.corda.notary.plugin.common.NotarisationResponse
+import com.r3.corda.notary.plugin.common.NotaryErrorGeneral
+import com.r3.corda.notary.plugin.common.NotaryErrorReferenceStateUnknown
+import com.r3.corda.notary.plugin.nonvalidating.api.NonValidatingNotarisationPayload
+import net.corda.crypto.testkit.SecureHashUtils
+import net.corda.ledger.utxo.flow.impl.timewindow.TimeWindowBetweenImpl
+import net.corda.uniqueness.datamodel.impl.UniquenessCheckErrorReferenceStateUnknownImpl
+import net.corda.uniqueness.datamodel.impl.UniquenessCheckResponseImpl
+import net.corda.uniqueness.datamodel.impl.UniquenessCheckResultFailureImpl
+import net.corda.uniqueness.datamodel.impl.UniquenessCheckResultSuccessImpl
+import net.corda.v5.application.crypto.DigitalSignatureAndMetadata
+import net.corda.v5.application.crypto.DigitalSignatureMetadata
+import net.corda.v5.application.crypto.DigitalSignatureVerificationService
+import net.corda.v5.application.membership.MemberLookup
+import net.corda.v5.application.messaging.FlowSession
+import net.corda.v5.application.serialization.SerializationService
+import net.corda.v5.base.types.MemberX500Name
+import net.corda.v5.crypto.DigitalSignature
+import net.corda.v5.ledger.utxo.StateAndRef
+import net.corda.v5.ledger.utxo.StateRef
+import net.corda.v5.ledger.utxo.transaction.UtxoLedgerTransaction
+import net.corda.v5.ledger.utxo.transaction.UtxoSignedTransaction
+import net.corda.v5.ledger.utxo.uniqueness.client.LedgerUniquenessCheckerClientService
+import net.corda.v5.membership.MemberInfo
+import net.corda.v5.serialization.SerializedBytes
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.mockito.Mockito
+import org.mockito.kotlin.any
+import org.mockito.kotlin.doAnswer
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.doThrow
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.whenever
+import java.security.PublicKey
+import java.time.Instant
+
+class NonValidatingNotaryServerFlowImplTest {
```

Comment:
```suggestion
@TestInstance(TestInstance.Lifecycle.PER_CLASS)
class NonValidatingNotaryServerFlowImplTest {
```

If this works for the use case here, this should speed up the test runs: https://phauer.com/2018/best-practices-unit-testing-kotlin/#avoid-static-and-reuse-the-test-class-instance